File: 1635-number-of-good-pairs/number-of-good-pairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        seen, count = {}, 0
        for num in nums:
            seen[num] = seen.get(num, 0) + 1
        for key in seen:
            count += comb(seen[key], 2)
        return count

File: 1635-number-of-good-pairs/number-of-good-pairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        int count = 0;
        HashMap<Integer, Integer> seen = new HashMap<>();
        for (int num : nums) {
            int temp = seen.getOrDefault(num, 0);
            count += temp;
            seen.put(num, temp + 1);
        }
        return count;
    }
}